nixos/modules/services/networking/tailscale-cert.nix

```diff
@@ -0,0 +1,143 @@
+      type = lib.types.bool;
+      default = false;
+      description = "Whether to configure the Nginx virtual host matching <option>services.tailscale-cert.domain</option> with the Tailscale certificate and key";
+    };
+  };
+
+  config = lib.mkIf cfg.enable (lib.mkMerge [
+    {
+      systemd.services.tailscale-cert = {
+        description = "Tailscale cert service";
+        requires = [ "network-online.target" ];
+        after = [ "network-online.target" ];
+        serviceConfig = {
+          Type = "oneshot";
+
+          StateDirectory = cfg.dataDir;
+          ProtectHome = true;
+          PrivateTmp = true;
+          PrivateDevices = true;
+          ProtectHostname = true;
+          ProtectClock = true;
+          ProtectKernelTunables = true;
+          ProtectKernelModules = true;
+          ProtectKernelLogs = true;
+          ProtectControlGroups = true;
+          RestrictAddressFamilies = [ "AF_UNIX" "AF_INET" "AF_INET6" ];
+          RestrictNamespaces = true;
+          LockPersonality = true;
+          MemoryDenyWriteExecute = true;
+          RestrictRealtime = true;
+          RestrictSUIDSGID = true;
+          RemoveIPC = true;
+          PrivateMounts = true;
+          ProtectSystem = "strict";
+          SystemCallFilter = [ "@system-service" ];
+          NoNewPrivileges = true;
+
+          ExecStart = toString [
+            "${lib.getExe cfg.package} cert"
+            "--cert-file /var/lib/${cfg.dataDir}/${cfg.certFile}"
+            "--key-file /var/lib/${cfg.dataDir}/${cfg.keyFile}"
+            (lib.optionalString (cfg.minValidity != null) "--min-validity ${cfg.minValidity}")
+            cfg.domain
+          ];
+        };
+      };
+
+      systemd.timers.tailscale-cert = {
+        description = "Fetch HTTPS certificates with `tailscale cert`";
+        wantedBy = [ "timers.target" ];
+        timerConfig = {
+          Unit = "tailscale-cert.service";
+          # Ensure timer starts when first activated
+          OnActiveSec = 0;
+          # Ensure timer runs at regular intervals
+          OnCalendar = cfg.frequency;
+        };
+      };
+    }
+    (lib.mkIf cfg.configureNginx (
+      let
+        certFile = "/var/lib/${cfg.dataDir}/${cfg.certFile}";
+        keyFile = "/var/lib/${cfg.dataDir}/${cfg.keyFile}";
+        nginxUser = config.services.nginx.user;
+        nginxGroup = config.services.nginx.group;
+      in
+      {
+        services.nginx = {
+          enable = lib.mkDefault true;
+
+          virtualHosts."${cfg.domain}" = {
+            sslCertificate = certFile;
+            sslCertificateKey = keyFile;
+          };
+        };
+        systemd.services.nginx.serviceConfig.ReadOnlyPaths = [ certFile keyFile ];
+        systemd.services.tailscale-cert = {
+          requiredBy = [ "nginx.service" ];
+          before = [ "nginx.service" ];
+          serviceConfig.ExecStartPost = ''
+            ${pkgs.coreutils}/bin/chown ${nginxUser}:${nginxGroup} ${certFile} ${keyFile}
```

Comment:
I think we shouldn't need this chown if we would pass in the certs file and key via LoadCredentials?

Reply:
`LoadCredential` seems pretty cool, but I don't know how to then pass `$CREDENTIALS_DIRECTORY/cert` to `sslCertificate`, because it expects a path. I guess I could use `extraConfig` instead, but is it worth it? 

What's the benefit of using `LoadCredential`, besides looking cleaner?

Reply:
This is how far I've gotten:

```nix
    (lib.mkIf cfg.configureNginx ({
      services.nginx = {
        enable = lib.mkDefault true;

        virtualHosts."${cfg.domain}" = {
          sslCertificate = "$CREDENTIALS_DIRECTORY/cert";
          sslCertificateKey = "$CREDENTIALS_DIRECTORY/key";
        };
      };
      systemd.services.nginx.serviceConfig.LoadCredential = [
        "cert:/var/lib/${cfg.dataDir}/${cfg.certFile}"
        "key:/var/lib/${cfg.dataDir}/${cfg.keyFile}"
      ];
      systemd.services.tailscale-cert = {
        requiredBy = [ "nginx.service" ];
        before = [ "nginx.service" ];
      };
    }
```